### DISCUSSION TITLE
Datepicker doesn't hide on outside click when using custom overlay

### DISCUSSION THREAD

**Discussion by peterbartels** [gpbl/react-day-picker#1120] [Support]
Try it here:
http://react-day-picker.js.org/examples/input-custom-overlay

- Click on the textbox to open the datepicker
- Click on arrow-right (to go to the next month)
- click outside the datepicker
- it doesn't close (only when selecting a date)

Same problem occurs when using a custom navbar

**Comment by matthova** [comment]
Ran into the same problem. Going to do a bit of digging to see if I can find out why, but let me know if you've found a workaround

**Comment by matthova** [comment]
PR: https://github.com/gpbl/react-day-picker/pull/871

**Comment by matthova** [comment]
As described in #871 I'm not sure how to patch the bug while maintaining standard behavior.
Instead  I'm going to create a fork where `showOverlay` is driven by props.

**Comment by matthova** [comment]
Here's a working implementation by driving showOverlay state from props: https://github.com/gpbl/react-day-picker/pull/872

**Comment by silviu-at** [comment]
Hi there! I ran into the same issue when providing a custom input component to the `DayPickerInput` component.

@matthova Thanks for looking closer into this, I can see you tried a couple of implementations.

Are there any plans to address this issue? What about known workarounds? In my opinion, this is quite a serious problem as I can imagine a considerable amount of users will want to provide their own input or overlay components.

**Comment by matthova** [comment]
@TSilviu the custom implementation shown in the examples in #872 has been working great for us.

**Comment by MuhammadSaadQadeer** [comment]
Is this fixed ? @matthova I tried your custom example but its not working for me, any insights on this ? Also I am using this with formik
